File: backend/app/schemas/inspection.py
import uuid
from datetime import date, datetime
from typing import Any

from pydantic import BaseModel

from app.models.inspection import InspectionType, OverallCondition


class InspectionCreate(BaseModel):
    lease_id: uuid.UUID | None = None
    property_id: uuid.UUID | None = None
    inspection_type: InspectionType
    inspection_date: date
    inspector_name: str | None = None
    tenant_present: bool = True
    overall_condition: OverallCondition | None = None
    notes: str | None = None
    rooms_data: dict[str, Any] | None = None


class InspectionUpdate(BaseModel):
    inspection_date: date | None = None
    inspector_name: str | None = None
    tenant_present: bool | None = None
    overall_condition: OverallCondition | None = None
    notes: str | None = None
    rooms_data: dict[str, Any] | None = None


class InspectionResponse(BaseModel):
    id: uuid.UUID
    lease_id: uuid.UUID | None = None
    property_id: uuid.UUID | None = None
    inspection_type: InspectionType
    inspection_date: date
    inspector_name: str | None = None
    tenant_present: bool
    overall_condition: OverallCondition | None = None
    notes: str | None = None
    rooms_data: dict[str, Any] | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class InspectionListResponse(BaseModel):
    items: list[InspectionResponse]
    total: int
    skip: int
    limit: int

File: frontend/src/types/inspection.ts
export type InspectionType = 'entree' | 'sortie' | 'contradictoire' | 'periodique'
export type OverallCondition = 'tres_bon' | 'bon' | 'moyen' | 'mauvais'

export const INSPECTION_TYPE_LABELS: Record<InspectionType, string> = {
  entree: 'État des lieux d\'entrée',
  sortie: 'État des lieux de sortie',
  contradictoire: 'Contradictoire',
  periodique: 'Visite périodique',
}

export const CONDITION_LABELS: Record<OverallCondition, string> = {
  tres_bon: 'Très bon',
  bon: 'Bon',
  moyen: 'Moyen',
  mauvais: 'Mauvais',
}

export const CONDITION_VARIANTS: Record<OverallCondition, 'green' | 'blue' | 'yellow' | 'red'> = {
  tres_bon: 'green',
  bon: 'blue',
  moyen: 'yellow',
  mauvais: 'red',
}

export interface Inspection {
  id: string
  lease_id?: string
  property_id?: string
  inspection_type: InspectionType
  inspection_date: string
  inspector_name?: string
  tenant_present: boolean
  overall_condition?: OverallCondition
  notes?: string
  rooms_data?: Record<string, unknown>
  created_at: string
  updated_at: string
}

export interface InspectionListResponse {
  items: Inspection[]
  total: number
  skip: number
  limit: number
}
